portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,10 +39,11 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxComments = Integer.parseInt(request.getParameter("num"));
```

Comment:
As it's a parameter for the max comments, I'd call it that as well rather than "num" (which doesn't really suggest what it's purpose is).